lib/resolver.go

```diff
@@ -0,0 +1,82 @@
+package vegeta
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync/atomic"
+)
+
+type resolver struct {
+	addresses []string
+	dialer    *net.Dialer
+	idx       uint64
+}
+
+// NewResolver - create a new instance of a dns resolver for plugging
+// into net.DefaultResolver.  Addresses should be a list of
+// ip addresses and optional port numbers, separated by colon.
+// For example: 1.2.3.4:53 and 1.2.3.4 are both valid.  In the absence
+// of a port number, 53 will be used instead.
+func NewResolver(addresses []string) (*net.Resolver, error) {
```

Comment:
This should not be public. It’s not core to Vegeta as a library. Also, isn’t there any library out there that does this exactly?

Reply:
Not that I'm aware of.  Would you prefer this be moved into main?  Or would you prefer it be moved into a separate subpackage?